test/extension-tests/single-root-UI/test/envs-presets.test.ts

```diff
@@ -15,9 +15,11 @@ suite('Environment Variables in Presets', () => {
         testEnv = new DefaultEnvironment('test/extension-tests/single-root-UI/project-folder', build_loc, exe_res);
         testEnv.projectFolder.buildDirectory.clear();
 
-        await vscode.commands.executeCommand('cmake.setConfigurePreset', 'LinuxUser1');
+        await vscode.commands.executeCommand('cmake.setConfigurePreset', process.platform === 'win32' ? 'WindowsUser1' : 'LinuxUser1');
         await vscode.commands.executeCommand('cmake.setBuildPreset', '__defaultBuildPreset__');
         await vscode.commands.executeCommand('cmake.setTestPreset', '__defaultTestPreset__');
+
+        await vscode.workspace.getConfiguration('cmake', vscode.workspace.workspaceFolders![0].uri).update('useCMakePresets', 'always');
```

Comment:
was this the main change? meaning that we needed to update the presets and then set the 'useCMakePresets' to 'always' so that the env variables are available on time for the tests?
Because I would normally think that the setup will run before the tests run and we will have the env variables available by that point Line 45).

Reply:
This is done because we can't currently await the work that happens after the setting change.  If that work is still running when we change the presets, the preset change will fail and that's what's currently causing the test failure.